internal/pkg/views/treeview.go

```diff
@@ -71,11 +72,27 @@ func (w *ListWidget) Layout(g *gocui.Gui) error {
 	linesPerItem := linesUsedCount / len(w.items)
 	maxItemsCanShow := (w.h / linesPerItem) - 1 // minus 1 to be on the safe side
 
-	for i, item := range allItems {
-		// Skip items above the selection to allow scrolling
-		if w.selected > maxItemsCanShow && i < w.selected {
-			continue
-		}
+	topIndex := w.lastTopIndex
+	bottomIndex := w.lastTopIndex + maxItemsCanShow
+
+	if w.selected >= bottomIndex {
+		// need to adjust down
+		diff := w.selected - bottomIndex + 1
+		topIndex += diff
+		bottomIndex += diff
+	}
+	if w.selected < topIndex {
+		// need to adjust up
+		diff := topIndex - w.selected
+		topIndex -= diff
+		bottomIndex -= diff
+	}
+	w.lastTopIndex = topIndex
+	if bottomIndex > len(allItems) {
+		bottomIndex = len(allItems) - 1
+	}
+
+	for _, item := range allItems[topIndex : bottomIndex+1] {
```

Comment:
I suspect that this line is trying to be too clever and failing if there are no list items

Reply:
Yeah that makes sense to me based on the stacktrace. Sorry I didn't get a the repro as was just randomly jumping around in a sub without `--debug` on and then couldn't find the RG I opened :(

Reply:
I didn't manage to repro (tried creating a new RG and viewing deployments) but I've tweaked this bit of code to hopefully be more resilient